src/main/java/com/atlassian/db/replica/internal/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -0,0 +1,54 @@
+package com.atlassian.db.replica.internal.aurora;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+
+import static com.atlassian.db.replica.internal.aurora.AuroraEndpoints.instanceEndpoint;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * It allows to pull Aurora Replicas cluster information
+ */
+public final class AuroraReplicasDiscoverer {
+    private final Connection connection;
+
+    public AuroraReplicasDiscoverer(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * It provides list of replica endpoints based on reader endpoint and server ids
+     * @param readerEndpoint - Aurora reader endpoint
+     * @return list of replicas endpoints
+     * @throws SQLException
+     */
+    public List<AuroraEndpoint> fetchReplicasEndpoints(String readerEndpoint) throws SQLException {
+        return fetchReplicasServerIds()
+            .stream()
+            .map(serverId -> instanceEndpoint(readerEndpoint, serverId))
+            .collect(toList());
+    }
+
+    /**
+     * Asks Aurora about the list of server ids
+     * @return server ids
+     * @throws SQLException
+     */
+    public List<String> fetchReplicasServerIds() throws SQLException {
+        LinkedList<String> ids = new LinkedList<>();
+        try (ResultSet rs =
+                 connection.prepareStatement(
+                     "SELECT server_id FROM aurora_global_db_instance_status() WHERE session_id != 'MASTER_SESSION_ID'")
```

Comment:
I wonder whether aurora specific functions have to be used, or postgress would be also compatible ?
If so I would be for postgress, why ? aaurora is postgress compatible and it can be easier test on postgres-docker

Reply:
```    
assertThat(executeQueryAndReturnWithMetaData(postgressProvider.getPureMainConnection(), "SELECT pg_is_in_recovery()")).isEqualTo("f pg_is_in_recovery");
assertThat(executeQueryAndReturnWithMetaData(postgressProvider.getPureReplicaConnection(), "SELECT pg_is_in_recovery()")).isEqualTo("t pg_is_in_recovery");
```

`select pg_backend_pid();`

https://www.postgresql.org/message-id/BANLkTimwRLFOPriKoUyG6DESTTLBOa_N3Q@mail.gmail.com

https://www.postgresql.org/docs/9.1/functions-info.html